<launch>
  <arg name="configuration_directory" default="$(find campusrover_cartographer)/launch/configs"/>
  <arg name="configuration_basenames" default="assets_3d_pcd.lua"/>
  <arg name="urdf_filename" default="$(find campusrover_description)/urdf/campusrover.urdf"/>
  <arg name="pose_graph_filename" default="$(env HOME)/bags/campusrover_bag/20210819_ntut.bag.pbstream" />
  <arg name="bag_filenames" default="$(env HOME)/bags/campusrover_bag/20210819_ntut.bag"/>

  <node name="cartographer_assets_writer" pkg="cartographer_ros" required="true"
      type="cartographer_assets_writer" args="
          -configuration_directory $(arg configuration_directory)
          -configuration_basename $(arg configuration_basenames)
          -urdf_filename $(arg urdf_filename)
          -bag_filenames $(arg bag_filenames)
          -pose_graph_filename $(arg pose_graph_filename)"
      output="screen">
  </node>
</launch>
